spreadsheet/formula/prefixhorizontalrange.go

```diff
@@ -58,3 +60,19 @@ func (r PrefixHorizontalRange) Reference(ctx Context, ev Evaluator) Reference {
 	pfx := r.pfx.Reference(ctx, ev)
 	return Reference{Type: ReferenceTypeHorizontalRange, Value: r.horizontalRangeReference(pfx.Value)}
 }
+
+// String returns a string representation of a horizontal range with prefix.
+func (r PrefixHorizontalRange) String() string {
+	return fmt.Sprintf("%s!%d:%d", r.pfx.String(), r.rowFrom, r.rowTo)
+}
+
+// Update updates references in the PrefixHorizontalRange after removing a row/column.
+func (r PrefixHorizontalRange) Update(q *update.UpdateQuery) Expression {
+	switch q.UpdateType {
```

Comment:
switch statement looks a bit redundant when all the cases just return r
Could be made more complex when needed instead